homeassistant/components/shelly/utils.py

```diff
@@ -353,18 +354,33 @@ def is_rpc_momentary_input(
     return cast(bool, config[key]["type"] == "button")
 
 
-def is_block_channel_type_light(settings: dict[str, Any], channel: int) -> bool:
+def is_block_channel_type_light(settings: dict[str, Any], channel: str) -> bool:
     """Return true if block channel appliance type is set to light."""
-    app_type = settings["relays"][channel].get("appliance_type")
+    app_type = settings["relays"][int(channel)].get("appliance_type")
     return app_type is not None and app_type.lower().startswith("light")
 
 
-def is_rpc_channel_type_light(config: dict[str, Any], channel: int) -> bool:
+def is_block_exclude_from_relay(settings: dict[str, Any], block: Block) -> bool:
+    """Return true if block should be excluded from switch platform."""
+
+    if settings.get("mode") == "roller":
+        return True
+
+    if TYPE_CHECKING:
+        assert block.channel is not None
+
+    return is_block_channel_type_light(settings, block.channel)
+
+
+def is_rpc_channel_type_light(
+    config: dict[str, Any], status: dict[str, Any], channel: str
```

Comment:
status is unused here

Reply:
I think you can avoid utils changes and the light changes with something like https://github.com/bdraco/home-assistant/commit/3987338e93aa8d088d2dad150658b23a7e8d8d85

Reply:
> status is unused here

but is not on other `removal_condition`, this is why is there

> I think you can avoid utils changes and the light changes with something like [bdraco@3987338](https://github.com/bdraco/home-assistant/commit/3987338e93aa8d088d2dad150658b23a7e8d8d85)

Probably, but changes are there as the `removal_condition` will be more complex once all platforms will move to **EntityDescription**

Reply:
per discord discussion, will push it in. We can refactor it some more in the next PR